tools/test/toolchains/test_toolchains.py

```diff
@@ -127,38 +168,63 @@ def test_gcc_minimal_printf(self):
         for i in minimal_printf_wraps:
             self.assertIn(i, gcc_obj.flags["ld"])
 
-    def test_gcc_arm_default_lib(self):
+    def test_gcc_arm_c_lib(self):
         """Test that linker flags are correctly added to an instance of GCC_ARM."""
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
         mock_target.supported_c_libs = {"gcc_arm": ["small"]}
-        mock_target.default_lib = "sMALL"
+        mock_target.c_lib = "sMALL"
         mock_target.supported_toolchains = ["GCC_ARM"]
         mock_target.is_TrustZone_secure_target = False
         gcc_arm_obj = GCC_ARM(mock_target)
         self.assertIn("-DMBED_RTOS_SINGLE_THREAD", gcc_arm_obj.flags["common"])
         self.assertIn("-D__NEWLIB_NANO", gcc_arm_obj.flags["common"])
         self.assertIn("--specs=nano.specs", gcc_arm_obj.flags["ld"])
 
-    def test_gcc_arm_default_lib_std_exception(self):
+    def test_gcc_arm_c_lib_std_exception(self):
         """Test that an exception is raised if the std C library is not supported for a target on the GCC_ARM toolchain."""
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
         mock_target.default_toolchain = "ARM"
-        mock_target.default_lib = "std"
-        mock_target.supported_c_libs = {"arm": ["small"]}
-        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+        mock_target.c_lib = "std"
+        mock_target.supported_c_libs = {"gcc_arm": ["small"]}
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.c_lib)):
             GCC_ARM(mock_target)
 
-    def test_gcc_arm_default_lib_small_exception(self):
+    def test_gcc_arm_c_lib_small_exception(self):
         """Test that an exception is raised if the small C library is not supported for a target on the GCC_ARM toolchain."""
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
-        mock_target.default_lib = "small"
-        mock_target.supported_c_libs = {"arm": ["std"]}
+        mock_target.c_lib = "small"
+        mock_target.supported_c_libs = {"gcc_arm": ["std"]}
         mock_target.default_toolchain = "ARM"
         mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5"]
-        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.c_lib)):
+            GCC_ARM(mock_target)
+
+    def test_gcc_arm_selected_c_libs(self):
```

Comment:
The name is not very precise. I would use:
```suggestion
    def test_gcc_arm_small_lib_selection(self):
```